frappe/website/utils.py

```diff
@@ -117,7 +117,12 @@ def _get_home_page():
 			home_page = frappe.db.get_single_value("Website Settings", "home_page")
 
 		if not home_page:
-			home_page = "login" if frappe.session.user == "Guest" else "me"
+			if frappe.session.user == "Guest":
+				home_page = "login"
+			elif frappe.session.data.user_type == "System User":
+				home_page = "app"
```

Comment:
This is kind of a problem right? anyone who has homepage configured on role, portal will now start seeing that instead


The current logic, in order of importance:

- If user has ANY role that has homepage setup
- Portal setting's default homepage
- use hook called "get_website_user_home_page"
- use website setting homepage
- fallback to `/app` <--- this is way too many levels down for average user to figure out what they need to change to make their homepage back to `/app`

Perheps we should add a separate config for system users. 

https://github.com/frappe/frappe/blob/4671460b650f6054ff21191f8012bd221eae84dd/frappe/website/utils.py#L99-L117

Reply:
> Perhaps we should add a separate config for system users.

Wanted to avoid this.

> This is kind of a problem right? anyone who has homepage configured on role, portal will now start seeing that instead

This is how it should have been?

If you look at resolve_path, the home page is computed for both system users and website users.

https://github.com/frappe/frappe/blob/c0c5b2ebdddbe8898ce2d5e5365f4931ff73b6bf/frappe/website/path_resolver.py#L125-L140

So there is inconsistency in how `get_home_page` is called during `/login` and `/index`